File: src/ambilight/Ambilight.java
package ambilight;


public interface Ambilight {

	void init(int ledsWidth, int ledsHeight, int leds[][]);

	byte[][] getScreenSegmentsColors();


	default void benchmarkOnce() {
		long start = System.currentTimeMillis();
		getScreenSegmentsColors();
		long end = System.currentTimeMillis();

		long duration = end - start;

		System.out.println("Time spent: " + duration + " ms.");
	}

	default void benchmarkRepeatedly(int ms) {
		long start = System.currentTimeMillis();
		int counter = 0;
		while (true) {
			long current = System.currentTimeMillis();
			if (current - start > ms) {
				break;
			}
			getScreenSegmentsColors();
			counter++;
		}
		System.out.println("During " + ms + "ms, called " + counter + " times.");
	}

}

File: core/src/main/java/me/khol/ambilight/Ambilight.kt
package me.khol.ambilight

abstract class Ambilight {

	abstract fun getScreenSegmentsColors(): Array<LedColor>

	fun benchmarkOnce() {
		val start = System.currentTimeMillis()
		getScreenSegmentsColors()
		val end = System.currentTimeMillis()

		val duration = end - start

		println("Time spent: $duration ms.")
	}

	fun benchmarkRepeatedly(ms: Int) {
		val start = System.currentTimeMillis()
		var counter = 0
		while (true) {
			val current = System.currentTimeMillis()
			if (current - start > ms) {
				break
			}
			getScreenSegmentsColors()
			counter++
		}
		println("During " + ms + "ms, called " + counter + " times.")
	}
}
